### PR TITLE
Markdownlinkconverter

### PR DESCRIPTION
**Pull request by KurtMi**

### CODE REVIEW

File: doc/markdownlinkconverter/README.md [hunk 1]
```diff
@@ -29,7 +29,5 @@ happens in 2 passes, which is needed because there can be files with no title.
 
 ## Further improvements (which will be introduced in a later version):
 
-* redirect links to code files right
 * optimize pdf output (also UTF-8 encoding)
-* valid link checker
```
Comment (markus2330) [inline doc/markdownlinkconverter/README.md]: Can you describe the link checker a bit in the docu?

File: doc/markdownlinkconverter/markdownlinkconverter.c [hunk 2]
```diff
@@ -60,6 +64,7 @@ struct transitionTitle { int t[10][4]; };
  */
 
 const int linkStart = 0;
+int lineCount = 0;
```
Comment (markus2330) [inline doc/markdownlinkconverter/markdownlinkconverter.c]: If possible this should be avoided. For this kind of tools its ok, but it is definitely a code smell.

File: doc/markdownlinkconverter/markdownlinkconverter.c [hunk 3]
```diff
@@ -394,34 +400,50 @@ static void printTarget(FILE * output, char * target, char * filenameInElektra)
...
+	// if target ends with /, link it to README.md
+	if (target[strlen (target) - 1] == '/')
+	{
+		strcpy (&pathToLink[strlen (pathToLink)], "README.md");
+	}
+	//validate link
+	FILE * test = fopen (pathToLink, "r");
+	if (!test)
+	{
+		fprintf (stderr, "INVALID LINK %s:%i    \"%s\"\n", inputFilename, lineCount, backupTarget);
```
Comment (markus2330) [inline doc/markdownlinkconverter/markdownlinkconverter.c]: Is this syntax compatible to gcc warning/errors? (Do Editors/IDE jump to the correct line?)

**Comment by markus2330** [comment]
Very nice! Thank you a lot!

File: doc/markdownlinkconverter/markdownlinkconverter.c [hunk 4]
```diff
@@ -394,34 +400,50 @@ static void printTarget(FILE * output, char * target, char * filenameInElektra)
...
 		++target; // remove starting /
+		strcpy (&pathToLink[indexofElektraRoot], target);
+	}
+	// if target ends with /, link it to README.md
+	if (target[strlen (target) - 1] == '/')
+	{
+		strcpy (&pathToLink[strlen (pathToLink)], "README.md");
+	}
+	//validate link
+	FILE * test = fopen (pathToLink, "r");
```
Comment (markus2330) [inline doc/markdownlinkconverter/markdownlinkconverter.c]: stat would be a bit faster and no need to close the file afterwards. But it is not as portable. So fopen is ok.

File: doc/markdownlinkconverter/markdownlinkconverter.c (hunk 3 above)
Comment (KurtMi) [inline doc/markdownlinkconverter/markdownlinkconverter.c]: I am not sure what you mean

Comment (KurtMi) [inline doc/markdownlinkconverter/markdownlinkconverter.c]: the lineCount is not so precise (+/-1 Line) and the syntax is like `git grep`

Comment (markus2330) [inline doc/markdownlinkconverter/markdownlinkconverter.c]: Ok, I tested it, it does not work.

E.g. if gcc prints to stderr:
```
/e/src/plugins/ini/ini.c|500 col 13| warning: ‘outputDebug’ defined but not used [-Wunused-function]
```
my editor recognize that the warning is in line 500, column 13. Thus I can directly jump to that point.
The INVALID LINK... does not match the pattern my editor uses, thus it is printed, but the editor cannot jump to the place.

Is it possible that you also print the invalid links in such a format:
```
/e/src/plugins/ini/ini.c|500 col 13| warning: invalid link
```

Comment (KurtMi) [inline doc/markdownlinkconverter/markdownlinkconverter.c]: beside the 'INVALID LINK' part

Comment (KurtMi) [inline doc/markdownlinkconverter/markdownlinkconverter.c]: yes but the `col' part is not possible.

File: doc/markdownlinkconverter/README.md (hunk 1 above)
Comment (KurtMi) [inline doc/markdownlinkconverter/README.md]: done

**Comment by markus2330** [comment]
Can you ping me if you are ready? From my side, the PR looks great.

**Comment by KurtMi** [comment]
@markus2330 finished!

**Comment by markus2330** [comment]
funny macro name, I will wait for build server and merge then

**Comment by markus2330** [comment]
Thank you for your nice work!

**Pull request merged**